DOCS-11355 Gmail 1.0 (#1542)
mulesoft/docs-connectors: gmail/1.0/modules/ROOT/pages/gmail-connector-studio.adoc

To configure the global element for Gmail Connector, follow these steps:
. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. On the *General* tab, On the *General* tab, in *Connection*, select the <<oauth2c>> authentication method to configure:

+
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[oauth2c]]
=== Oauth2c

Enter the following information on the *General* tab of the *Global Element Properties* screen to configure Oauth2c authentication:

> [%header,cols=30s,70a]

Reviewer: Add a line between intro sentence and table.

include::_attributes.adoc[]
endif::[]

Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.

To add and configure a connector in Studio:

. <<create-mule-project,Create a Mule project>>.
. <<add-connector-to-project,Add the connector to your Mule project>>.
. <<configure-source,Configure a source for the connector's flow>>.
. <<add-connector-operation,Add a connector operation to the flow>>.
. <<configure-global-element,Configure a global element for the connector>>.
. <<configure-other-fields,Configure the other connector fields>>.

When you run the connector, you can view the app log to check for problems in real time, as described in <<view-app-log,View the App Log>>.

If you are new to configuring connectors in Studio, see xref:connectors::introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]. If, after reading this topic, you need additional information about the connector fields, see the xref:gmail-connector-reference.adoc[Gmail Reference].

[[create-mule-project]]
== Create a Mule Project

In Studio, create a new Mule project in which to add and configure the connector:

> . In Studio, select *File > New > Mule Project*.

Reviewer: Hhhhhhmmmmmm It still looks like the line is missing.

. Enter a name for your Mule project and click *Finish*.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add Gmail Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and add the required dependencies to the project's `pom.xml` file:

. In the *Mule Palette* view, click *(X) Search in Exchange*.

> . In the *Add Dependencies to Project* window, type `Gmail` in the search field.

Author: ```suggestion
. In the *Add Dependencies to Project* window, type `gmail` in the search field.
```

> . Click Gmail in *Available modules*.

Author: ```suggestion
. Click *Gmail* in *Available modules*.
```

. Drag *Listener* to the Studio canvas.
. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP Listener source in the app.
. On the *General* tab, specify connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[add-connector-operation]]
== Add a Connector Operation to the Flow

When you add a connector operation to your flow, you are specifying an action for that connector to perform.
To add an operation for Gmail Connector, follow these steps:

. In the *Mule Palette* view, select *Gmail* and then select the desired operation.
. Drag the operation onto the Studio canvas to the right of the input source.

[[configure-global-element]]
== Configure a Global Element for the Connector

> When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Gmail system. Gmail Connector supports <mention the types of authorization the connector supports>.

Reviewer: ```suggestion
When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Gmail system. Gmail Connector supports Oauth2c authentication.
```

To configure the global element for Gmail Connector, follow these steps:

. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.

> . On the *General* tab, On the *General* tab, in *Connection*, select the <<oauth2c>> authentication method to configure:

Author: ```suggestion
. On the *General* tab, in *Connection*, select the <<oauth2c>> authentication method to configure:
```

+
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].

. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[oauth2c]]
=== Oauth2c

Enter the following information on the *General* tab of the *Global Element Properties* screen to configure Oauth2c authentication:
[%header,cols=30s,70a]

> 

Reviewer: The table isn't rendering correctly. I think because the extra line isn't there.

|===
|Field |User Action
|Name |Enter the configuration name.
|Connection | Select *Oauth2c*.
|Consumer Key | OAuth consumer key, as registered with the service provider.
|Consumer Secret | OAuth consumer secret, as registered with the service provider.
|Listener Config | Configuration for the HTTP listener that listens for requests on the access token callback endpoint.
|Callback Path | Path of the access token callback endpoint.
|Authorize Path | Path of the HTTP endpoint that triggers the OAuth dance.
|===

[[configure-other-fields]]
== Configure Additional Connector Fields

> After you configure a global element for Gmail, configure the the other required fields for the connector. The required fields vary depending on which connector operation you use.

Author: ```suggestion
After you configure a global element for Gmail, configure the other required fields for the connector. The required fields vary depending on which connector operation you use.
```

= Use Anypoint Studio to Configure Gmail Connector - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.

To add and configure a connector in Studio:
. <<create-mule-project,Create a Mule project>>.
. <<add-connector-to-project,Add the connector to your Mule project>>.
. <<configure-source,Configure a source for the connector's flow>>.
. <<add-connector-operation,Add a connector operation to the flow>>.
. <<configure-global-element,Configure a global element for the connector>>.
. <<configure-other-fields,Configure the other connector fields>>.

When you run the connector, you can view the app log to check for problems in real time, as described in <<view-app-log,View the App Log>>.

If you are new to configuring connectors in Studio, see xref:connectors::introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]. If, after reading this topic, you need additional information about the connector fields, see the xref:gmail-connector-reference.adoc[Gmail Reference].

[[create-mule-project]]
== Create a Mule Project

> In Studio, create a new Mule project in which to add and configure the connector:

Reviewer: ```suggestion
In Studio, create a new Mule project in which to add and configure the connector:

```

Reviewer: Need to add a line between the intro line and the  steps

. In Studio, select *File > New > Mule Project*.
. Enter a name for your Mule project and click *Finish*.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

> Add Gmail Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and add the required dependencies to the project's `pom.xml` file:

Reviewer: Need to add a line between intro lines and lists. I don't know why it's rendering this way. Did you create these files from the script? If so, I might need to fix the script.

. In the *Mule Palette* view, click *(X) Search in Exchange*.
. In the *Add Dependencies to Project* window, type `Gmail` in the search field.
. Click Gmail in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

[[configure-source]]
== Configure a Source

A source initiates a flow when a specified condition is met.

You can configure one of these sources to use with Gmail Connector:

* HTTP Listener +
Initiates a flow each time it receives a request on the configured host and port

* Scheduler +
Initiates a flow when a time-based condition is met

> For example, to configure an HTTP Listener source, follow these steps:

Reviewer: Add a line

. In the *Mule Palette* view, select *HTTP > Listener*.
. Drag *Listener* to the Studio canvas.
. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP Listener source in the app.
. On the *General* tab, specify connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[add-connector-operation]]
== Add a Connector Operation to the Flow

When you add a connector operation to your flow, you are specifying an action for that connector to perform.

> To add an operation for Gmail Connector, follow these steps:

Reviewer: Add a line

. In the *Mule Palette* view, select *Gmail* and then select the desired operation.
. Drag the operation onto the Studio canvas to the right of the input source.

[[configure-global-element]]
== Configure a Global Element for the Connector

When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Gmail system. Gmail Connector supports <mention the types of authorization the connector supports>.

> To configure the global element for Gmail Connector, follow these steps:

Reviewer: Add a line

. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. On the *General* tab, On the *General* tab, in *Connection*, select the authentication method to configure:
* <link to authentication method config section>
* <link to authentication method config section>

+

> You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].

Reviewer: ```suggestion
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
```
